src/ucp/wireup/wireup.c

```diff
@@ -967,6 +1015,73 @@ ucp_wireup_get_reachable_mds(ucp_ep_h ep,
     key->reachable_md_map = dst_md_map;
 }
 
+void ucp_wireup_check_config_intersect(ucp_ep_h ep,
+                                       ucp_ep_config_key_t *new_key)
+{
+    ucp_worker_h worker                            = ep->worker;
+    ucp_lane_index_t reuse_lane_map[UCP_MAX_LANES] = { UCP_NULL_LANE };
+    ucp_ep_config_key_t *old_key;
+    ucp_lane_index_t lane, reuse_lane;
+    ucp_wireup_ep_t *cm_wireup_ep;
+
+    if (ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL) {
+        return;
+    }
+
+    old_key = &ucp_ep_config(ep)->key;
+
+    ucp_ep_config_lanes_intersect(old_key, new_key, reuse_lane_map);
+
+    if (ucp_ep_config_key_has_cm_lane(new_key)) {
+        ucs_assert(old_key->wireup_lane != UCP_NULL_LANE);
+
+        /* set the correct WIREUP MSG lane in case of CM */
+        if ((new_key->wireup_lane != UCP_NULL_LANE) &&
+            (UCP_NULL_LANE !=
+             (reuse_lane = ucp_ep_config_matched_lane(old_key,
+                                                      new_key->wireup_lane,
+                                                      reuse_lane_map)))) {
+            if (new_key->wireup_lane != reuse_lane) {
+                ep->uct_eps[new_key->wireup_lane] = ep->uct_eps[reuse_lane];
+                ep->uct_eps[reuse_lane]       = NULL;
+            }
+        } else if (UCP_NULL_LANE !=
+                   (reuse_lane = reuse_lane_map[old_key->wireup_lane])) {
+            new_key->wireup_lane             = reuse_lane;
+        } else {
+            cm_wireup_ep                = ucp_ep_get_cm_wireup_ep(ep);
+            new_key->wireup_lane        = new_key->cm_lane;
+            reuse_lane                  = old_key->wireup_lane;
+            /* save UCT EP from the previous WIREUP lane to the CM WIREUP
+             * auxiliary EP */
+            cm_wireup_ep->aux_ep        = ep->uct_eps[reuse_lane];
+            cm_wireup_ep->aux_rsc_index = old_key->lanes[reuse_lane].rsc_index;
+            /* reset the UCT EP from the previous WIREUP lane to not
+             * destroy it, since it's not needed anymore in the new
+             * configuration, but will be used for WIREUP MSG */
+            ep->uct_eps[reuse_lane]     = NULL;
+        }
+    }
+
+    /* Need to discard only old lanes that won't be used anymore in the new
+     * configuraton. Also, UCT EPs with the lane index >= old_key->num_lanes
+     * could be set in case of CM, we have to not reset them */
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+        reuse_lane = reuse_lane_map[lane];
+        if (reuse_lane == UCP_NULL_RESOURCE) {
+            if (ep->uct_eps[lane] != NULL) {
+                ucp_worker_discard_uct_ep(worker, ep->uct_eps[lane],
+                                          UCT_FLUSH_FLAG_LOCAL,
+                                          NULL, NULL);
+                ep->uct_eps[lane]   = NULL;
+            }
+        } else if (reuse_lane != lane) {
+            ep->uct_eps[reuse_lane] = ep->uct_eps[lane];
```

Comment:
can you rewrite some lane, since do everything in one ep?

Reply:
yes, need to store them in the separate array